include/seqan3/std/span

```diff
@@ -253,137 +253,137 @@ class span<span_tp, dynamic_extent> {
     static constexpr index_type extent = dynamic_extent;
 
 // [span.cons], span constructors, copy, assignment, and destructor
-    constexpr span() noexcept : data{nullptr}, size{0} {}
+    constexpr span() noexcept : data_{nullptr}, size_{0} {}
 
     constexpr span           (const span&) noexcept = default;
     constexpr span& operator=(const span&) noexcept = default;
 
-    constexpr span(pointer ptr, index_type count) : data{ptr}, size{count} {}
-    constexpr span(pointer f, pointer l) : data{f}, size{distance(f, l)} {}
+    constexpr span(pointer ptr, index_type count) : data_{ptr}, size_{count} {}
+    constexpr span(pointer f, pointer l) : data_{f}, size_{distance(f, l)} {}
 
     template <size_t span_sz>
-    inline constexpr span(element_type (&arr)[span_sz]) noexcept : data{arr}, size{span_sz} {}
+    inline constexpr span(element_type (&arr)[span_sz]) noexcept : data_{arr}, size_{span_sz} {}
 
     template <size_t span_sz>
-    inline constexpr span(array<value_type, span_sz>& arr) noexcept : data{arr.data()}, size{span_sz} {}
+    inline constexpr span(array<value_type, span_sz>& arr) noexcept : data_{arr.data()}, size_{span_sz} {}
 
     template <size_t span_sz>
-    inline constexpr span(const array<value_type, span_sz>& arr) noexcept : data{arr.data()}, size{span_sz} {}
+    inline constexpr span(const array<value_type, span_sz>& arr) noexcept : data_{arr.data()}, size_{span_sz} {}
 
     template <class container>
     inline constexpr span(container& c,
             enable_if_t<is_span_compatible_container<container, span_tp>::value, nullptr_t> = nullptr)
-        : data{std::data(c)}, size{(index_type) std::size(c)} {}
+        : data_{std::data(c)}, size_{(index_type) std::size(c)} {}
 
     template <class container>
     inline constexpr span(const container& c,
             enable_if_t<is_span_compatible_container<const container, span_tp>::value, nullptr_t> = nullptr)
-        : data{std::data(c)}, size{(index_type) std::size(c)} {}
+        : data_{std::data(c)}, size_{(index_type) std::size(c)} {}
 
     template <class OtherElementType, ptrdiff_t _OtherExtent>
     inline constexpr span(const span<OtherElementType, _OtherExtent>& other,
                        enable_if_t<
                           is_convertible_v<OtherElementType(*)[], element_type (*)[]>,
                           nullptr_t> = nullptr) noexcept
-        : data{other.get_data()}, size{other.get_size()} {}
+        : data_{other.data()}, size_{other.size()} {}
 
 //    ~span() noexcept = default;
 
     template <ptrdiff_t count>
     inline constexpr span<element_type, count> first() const noexcept
     {
-        static_assert(count >= 0, "Count must be >= 0 in span::first()");
```

Comment:
This is in an constexpr context, so it could be `static_asserts` (?)
So why change it?

Reply:
i was a little over-ambitious with removing the wrong asserts maybe